### PR TITLE
Added Mergemedia ...

### PR DESCRIPTION
**Pull request by midnightmadwalk**
User Friendly MergeMedia ...

### CODE REVIEW

File: plugins/mergemedia.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    },
+)
+async def mergesave_(message: Message):
+    """mergesave"""
+    # saving files in a separate folder.
+    await message.edit("`downloading ...`")
+    try:
+        o_o = await message.client.download_media(
+          message=message.reply_to_message,
+          file_name='merge/',
```
Comment (code-rgb) [inline plugins/mergemedia.py]: userge/xcache/merge

File: plugins/mergemedia.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        "header": "save file for {tr}merge",
+        "usage": "{tr} reply to [media] for saving it ",
+    },
+)
+async def mergesave_(message: Message):
+    """mergesave"""
+    # saving files in a separate folder.
+    await message.edit("`downloading ...`")
+    try:
+        o_o = await message.client.download_media(
```
Comment (code-rgb) [inline plugins/mergemedia.py]: o__o is a strange variable name

File: plugins/mergemedia.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+          message=message.reply_to_message,
+          file_name='merge/',
+          progress=progress,
+          progress_args=(
+            message,
+            "`Saving for further merge !`"
+          ),
+        )
+    except AttributeError:
```
Comment (code-rgb) [inline plugins/mergemedia.py]: Instead off attribute error you can also check if the media is downloadable
Like => if not message.photo or message.video:

File: plugins/mergemedia.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    else:
+        output_path = "merge/output." + a_a
+    # ffmpeg.
+    await message.edit("`🏃️🏃🏃 ffmpeg ...`")
+    logs_ = await runcmd(
+      f'''ffmpeg -f concat -safe 0 -i merge.txt -map 0 -c copy -scodec copy {output_path}'''
+    )
+    # upload.
+    try:
+        await upload(message, Path(output_path))
```
Comment (code-rgb) [inline plugins/mergemedia.py:87]: You could have just checked it with os.path.exists()

File: plugins/mergemedia.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    except (NullStreamError, InputStreamError):
+        await message.err("Something went south generating ffmpeg log file.")
+        await message.reply(logs_)
+    else:
+      await message.edit("`successfully merged ...`")
+    # cleanup.
+    await message.edit("`🤯😪 cleaning mess ...`", del_in=10)
+    shutil.rmtree("merge")
```
Comment (code-rgb) [inline plugins/mergemedia.py]: add ignore_errors=True in rmtree

**Review by code-rgb — CHANGES_REQUESTED** [review]

File: plugins/mergemedia.py (hunk 5 above)
Comment (midnightmadwalk) [reply plugins/mergemedia.py]: okay :D

File: plugins/mergemedia.py (hunk 1 above)
Comment (midnightmadwalk) [reply plugins/mergemedia.py:34]: okay :D

Comment (midnightmadwalk) [reply plugins/mergemedia.py]: Okay 👌

File: plugins/mergemedia.py (hunk 2 above)
Comment (midnightmadwalk) [reply plugins/mergemedia.py]: okay 😗😗

File: plugins/mergemedia.py (hunk 4 above)
Comment (midnightmadwalk) [reply plugins/mergemedia.py:87]: same name files and weird codecs threw errors so i added that except: to generate logs ... and see what happened in ffmpeg ...

Comment (code-rgb) [reply plugins/mergemedia.py:87]: Okay

File: plugins/mergemedia.py (hunk 3 above)
Comment (midnightmadwalk) [reply plugins/mergemedia.py]: okay 💣🙃🙂

File: plugins/mergemedia.py [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+@userge.on_cmd(
+    "mergeclear",
+    about={
+        "header": "Incase you saved wrong media",
+        "usage": "{tr}mergeclear",
+    },
+)
+async def mergeclear_(message: Message):
+    '''incase you saved wrong media.'''
```
Comment (code-rgb) [inline plugins/mergemedia.py]: use """xyz"""

File: plugins/mergemedia.py [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+        "header": "Merge Media.",
+        "usage": "perform {tr}merge after saving videos with {tr}mergesave",
+    },
+)
+async def merge_(message: Message):
+    """MergeMedia with FFmpeg"""
+    name_ = message.input_str
+    # preparing text file.
+    await message.edit("`🙂🙃 Preparing text file ...`")
+    x_x = codecs.open("merge.txt", "w+", "utf-8")
```
Comment (code-rgb) [inline plugins/mergemedia.py]: change x_x to fo or any other variable

File: plugins/mergemedia.py [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+    if not r:
+        await message.err("Reply To Media, dear.")
+    if not (
+      r.audio
+      or r.document
+      or r.video
+      or r.video_note
+      or r.voice
+    ):
+        await message.err("Not Supported Extension")
```
Comment (code-rgb) [inline plugins/mergemedia.py:39]: After this need should check the mime type of document or I will just a make a function in utils for that so remove r.document or comment it out for now

**Review by code-rgb — CHANGES_REQUESTED** [review]
ignore deepsouce it's ok

File: plugins/mergemedia.py (hunk 6 above)
Comment (midnightmadwalk) [reply plugins/mergemedia.py]: 🤠

File: plugins/mergemedia.py (hunk 7 above)
Comment (midnightmadwalk) [reply plugins/mergemedia.py]: 👽

File: plugins/mergemedia.py (hunk 8 above)
Comment (midnightmadwalk) [reply plugins/mergemedia.py:39]: commented it out ...

**Pull request merged**